string mode
float64 input_1
float64 input_2
float64 input_3
float64 input_4
